File: backend/walmart_api.py
import requests
import os
from walmart_auth import WalmartAuth

class WalmartAPI:
    def __init__(self, config_path):
        self.auth = WalmartAuth(config_path)
        self.base_url = "https://developer.api.walmart.com/api-proxy/service/affil"

    def get_taxonomy(self):
        endpoint = f"{self.base_url}/product/v2/taxonomy"
        headers = self._get_headers()
        response = requests.get(endpoint, headers=headers)
        return self._process_response(response)

    def search_products(self, query, limit=10):
        endpoint = f"{self.base_url}/product/v2/search"
        headers = self._get_headers()
        params = {"query": query, "numItems": limit}
        response = requests.get(endpoint, headers=headers, params=params)
        return self._process_response(response)

    def get_product_details(self, product_id):
        endpoint = f"{self.base_url}/product/v2/items/{product_id}"
        headers = self._get_headers()
        response = requests.get(endpoint, headers=headers)
        return self._process_response(response)

    def _get_headers(self):
        headers = self.auth.generate_auth_headers()
        headers["Content-Type"] = "application/json"
        headers["Accept"] = "application/json"
        return headers

    def _process_response(self, response):
        if response.status_code == 200:
            return response.json()
        print(f"Error {response.status_code}: {response.text}")
        return None

File: src/services/apis/walmartApi.js
import axios from 'axios';

const API_BASE_URL = 'http://127.0.0.1:5000/api';

export const searchProducts = async (query) => {
  try {
    const response = await axios.get(`${API_BASE_URL}/search`, {
      params: { query }
    });
    return response.data.items || [];
  } catch (error) {
    console.error('Walmart API error:', error);
    return [];
  }
};

export const getProductDetails = async (productId) => {
  try {
    const response = await axios.get(`${API_BASE_URL}/product/${productId}`);
    return response.data;
  } catch (error) {
    console.error('Walmart API error:', error);
    return null;
  }
};